tests/base.py

```diff
@@ -195,12 +197,22 @@ def setUp(self, assetstoreType=None, dropModels=True):
                 createGridFsAssetstore(name='Test', db=gridfsDbName)
         elif assetstoreType == 'gridfsrs':
             gridfsDbName = 'girder_test_%s_rs_assetstore_auto' % assetstoreName
-            mongo_replicaset.startMongoReplicaSet()
+            self.replicaSetConfig = mongo_replicaset.makeConfig()
+            mongo_replicaset.startMongoReplicaSet(self.replicaSetConfig)
             self.assetstore = self.model('assetstore'). \
                 createGridFsAssetstore(
                 name='Test', db=gridfsDbName,
                 mongohost='mongodb://127.0.0.1:27070,127.0.0.1:27071,'
                 '127.0.0.1:27072', replicaset='replicaset')
+        elif assetstoreType == 'gridfsshard':
+            gridfsDbName = 'girder_test_%s_shard_assetstore_auto' % assetstoreName
+            self.replicaSetConfig = mongo_replicaset.makeConfig(
+                port=27073, shard=True, sharddb=None)
+            mongo_replicaset.startMongoReplicaSet(self.replicaSetConfig)
+            self.assetstore = self.model('assetstore'). \
```

Comment:
Could we remove the bacskslash and move the function name up to this line for readability